test/collector.test.ts

```diff
@@ -0,0 +1,351 @@
+import { expect, use } from 'chai';
+import Collector from '../build/contracts/Collector.json';
+import TestToken from '../build/contracts/TestToken.json';
+import {
+    deployContract,
+    loadFixture,
+    MockProvider,
+    solidity
+} from 'ethereum-waffle';
+import { Wallet } from '@ethersproject/wallet';
+
+use(solidity);
+
+const FIRST_PARTNER_SHARE = 20;
+const SECOND_PARTNER_SHARE = 30;
+const THIRD_PARTNER_SHARE = 40;
+const FOURTH_PARTNER_SHARE = 10;
+
+const buildPartners = (partners: Wallet[], shares?: number[]) => {
+    return [
+        {
+            beneficiary: partners[0].getAddress(),
+            share: shares?.[0] ?? FIRST_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[1].getAddress(),
+            share: shares?.[1] ?? SECOND_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[2].getAddress(),
+            share: shares?.[2] ?? THIRD_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[3].getAddress(),
+            share: shares?.[3] ?? FOURTH_PARTNER_SHARE
+        }
+    ];
+};
+
+async function prepareAll(wallets: Wallet[]) {
+    const [
+        owner,
+        remainder,
+        partner1,
+        partner2,
+        partner3,
+        partner4,
+        utilWallet
+    ] = wallets;
+
+    const testToken = await deployContract(owner, TestToken);
+    const partners = buildPartners([partner1, partner2, partner3, partner4]);
+
+    const collector = await deployContract(owner, Collector, [
+        owner.address,
+        testToken.address,
+        partners,
+        remainder.address
+    ]);
+
+    return { collector, testToken, partners, remainder, utilWallet };
+}
```

Comment:
I think I'd rather these were separate functions each doing a single job only.

Reply:
Not sure about this. This function groups three single calls to other functions, two of them (deployContract) external functions, and avoid to perform this calls every time in multiple points of the code.